### PR TITLE
Draw fill extrusions directly into the main framebuffer

### PR DESCRIPTION
**Pull request by kkaefer**
PR for https://github.com/mapbox/mapbox-gl-native/issues/13984

### CODE REVIEW

File: src/mbgl/programs/fill_extrusion_program.cpp [hunk 1]
```diff
@@ -36,9 +36,10 @@ float lightIntensity(const EvaluatedLight& light) {
 }
 
 FillExtrusionProgram::LayoutUniformValues FillExtrusionProgram::layoutUniformValues(
-    mat4 matrix, const TransformState& state, const EvaluatedLight& light) {
+    mat4 matrix, const TransformState& state, const float opacity, const EvaluatedLight& light) {
```
Comment (alexshalamov) [inline src/mbgl/programs/fill_extrusion_program.cpp]: nit: `const float` => `float`

File: src/mbgl/renderer/paint_parameters.cpp [hunk 2]
```diff
@@ -72,8 +72,10 @@ gfx::DepthMode PaintParameters::depthModeForSublayer(uint8_t n, gfx::DepthMaskTy
     return gfx::DepthMode { gfx::DepthFunctionType::LessEqual, mask, { nearDepth, farDepth } };
 }
 
-gfx::DepthMode PaintParameters::depthModeFor3D(gfx::DepthMaskType mask) const {
-    return gfx::DepthMode { gfx::DepthFunctionType::LessEqual, mask, { 0.0, 1.0 } };
+gfx::DepthMode PaintParameters::depthModeFor3D() const {
+    return gfx::DepthMode{ gfx::DepthFunctionType::LessEqual,
+                           gfx::DepthMaskType::ReadWrite,
+                           { 0.0, depthRangeSize } };
```
Comment (alexshalamov) [inline src/mbgl/renderer/paint_parameters.cpp]: Question, `depthRangeSize` is [set after](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/renderer_impl.cpp#L428) 3D pass, near / far range would be `{0.0f, 0.0f}`, right?

Comment (kkaefer) [reply src/mbgl/renderer/paint_parameters.cpp]: Yeah, this is a misnomer. The "3D" pass doesn't actually render anything related to 3D anymore. It's only used for heatmap and hillshading preprocessing, but not for fill extrusions anymore (we should rename it). The `depthModeFor3D` function isn't called in 3D pass. I'll keep the name as is to match JS' naming.

Comment (alexshalamov) [reply src/mbgl/renderer/paint_parameters.cpp]: :+1: thanks for explanation

**Comment by 1ec5** [comment]
This change seems to have introduced a drawing order regression: #14642.

**Pull request merged**